tools/mage/dev.go

```diff
@@ -168,18 +159,50 @@ func (Dev) DBStart() error {
 	if err := execDockerCompose(append([]string{"up", "-d"}, devDatabases...)...); err != nil {
 		return err
 	}
-	var cerr error
-	if os.Getenv("CI") != "true" {
-		flags := dockerComposeFlags(append([]string{"exec", "postgres", "pg_isready"})...)
-		for i := 0; i < 15; i++ {
-			if _, cerr = sh.Exec(nil, nil, nil, "docker-compose", flags...); cerr == nil {
-				break
+
+	// When TimescaleDB starts for the first time, it restarts Postgres after initialization.
+	// Therefore, pg_isready may return a successful exit code during initialization, while the database shuts down
+	// shortly after. Therefore, the ready check goes in two cycles and only returns after 10 successive ready
+	// indications.
+	if mg.Verbose() {
+		fmt.Println("Waiting for Postgres to be ready")
+	}
+	flags := dockerComposeFlags("exec", "-T", "postgres", "pg_isready")
+nextCycle:
+	for i := 0; i < 2; i++ {
+		var (
+			wasReady        bool
+			successiveReady int
+		)
+		for j := 0; j < 30; j++ {
+			time.Sleep(time.Second)
+			_, err := sh.Exec(nil, nil, nil, "docker-compose", flags...)
+			isReady := err == nil
+			switch {
+			case wasReady && !isReady:
+				if mg.Verbose() {
+					fmt.Println("Postgres is not ready anymore")
+				}
+				continue nextCycle
```

Comment:
Shouldn't you reset the `successiveReady` counter here ? Otherwise it is not really successive.